planetscale/imports.go

```diff
@@ -0,0 +1,321 @@
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type DataImportSource struct {
+	HostName            string `json:"hostname"`
+	Database            string `json:"schema_name"`
+	Port                int    `json:"port"`
+	SSLMode             string `json:"ssl_mode"`
+	SSLVerificationMode ExternalDataSourceSSLVerificationMode
+	UserName            string `json:"username"`
+	Password            string `json:"password"`
+}
+
+type ExternalDataSourceSSLVerificationMode int
+
+const (
+	SSLModeDisabled ExternalDataSourceSSLVerificationMode = iota
+	SSLModePreferred
+	SSLModeRequired
+	SSLModeVerifyCA
+	SSLModeVerifyIdentity
+)
+
+func (sm ExternalDataSourceSSLVerificationMode) String() string {
+	switch sm {
+	case SSLModeDisabled:
+		return "disabled"
+	case SSLModePreferred:
+		return "preferred"
+	case SSLModeRequired:
+		return "required"
+	case SSLModeVerifyCA:
+		return "verify_ca"
+	default:
+		return "verify_identity"
+	}
+}
+
+type DataImportState int
+
+const (
+	DataImportPreparingDataCopy DataImportState = iota
+	DataImportPreparingDataCopyFailed
+	DataImportCopyingData
+	DataImportCopyingDataFailed
+	DataImportSwitchTrafficPending
+	DataImportSwitchTrafficRunning
+	DataImportSwitchTrafficCompleted
+	DataImportSwitchTrafficError
+	DataImportReverseTrafficRunning
+	DataImportReverseTrafficCompleted
+	DataImportReverseTrafficError
+	DataImportDetachExternalDatabaseRunning
+	DataImportDetachExternalDatabaseError
+	DataImportReady
+)
+
+func (d DataImportState) String() string {
+	switch d {
+	case DataImportPreparingDataCopy:
+		return "preparing to copy data from external database"
+	case DataImportPreparingDataCopyFailed:
+		return "failed to copy data from external database"
+	case DataImportCopyingData:
+		return "copying data from external database"
+	case DataImportCopyingDataFailed:
+		return "failed to copy data from external database"
+	case DataImportSwitchTrafficPending:
+		return "PlanetScale database is running in replica mode"
+	case DataImportSwitchTrafficRunning:
+		return "switching PlanetScale database to primary mode"
+	case DataImportSwitchTrafficError:
+		return "failed to switching PlanetScale database to primary mode"
+	case DataImportReverseTrafficRunning:
+		return "switching PlanetScale database to replica mode"
+	case DataImportReverseTrafficError:
+		return "failed to switching PlanetScale database to replica mode"
+	case DataImportDetachExternalDatabaseRunning:
+		return "detaching external database from  PlanetScale database"
+	case DataImportDetachExternalDatabaseError:
+		return "failed to detach external database from  PlanetScale database"
+	}
+
+	return "all done"
+}
+
+type DataImport struct {
+	ID          string `json:"id"`
+	ImportState DataImportState
+	State       string    `json:"state"`
+	Errors      string    `json:"import_check_errors"`
+	StartedAt   time.Time `json:"started_at"`
+	FinishedAt  time.Time `json:"finished_at"`
+	DeletedAt   time.Time `json:"deleted_at"`
```

Comment:
These are all nullable in the API, I believe.

```suggestion
	StartedAt   *time.Time `json:"started_at"`
	FinishedAt  *time.Time `json:"finished_at"`
	DeletedAt   *time.Time `json:"deleted_at"`
```